compiler/src/iree/compiler/Codegen/Dialect/GPU/Transforms/Transforms.cpp

```diff
@@ -923,13 +923,32 @@ struct LowerShuffleTensor
                                 PatternRewriter &rewriter) const final {
     Location loc = shuffleOp.getLoc();
 
+    Value dest = shuffleOp.getDest();
+
+    // If the destination is a tensor.empty, replace it with an alloc_tensor.
+    if (auto emptyDest = shuffleOp.getDest().getDefiningOp<tensor::EmptyOp>()) {
+      Attribute sharedMemoryAddrSpace = gpu::AddressSpaceAttr::get(
+          rewriter.getContext(), gpu::GPUDialect::getWorkgroupAddressSpace());
+      auto allocTensor = rewriter.create<bufferization::AllocTensorOp>(
+          emptyDest->getLoc(), emptyDest->getResultTypes()[0],
+          emptyDest.getDynamicSizes());
+      allocTensor.setMemorySpaceAttr(sharedMemoryAddrSpace);
+      dest = allocTensor.getResult();
+    }
+
+    if (!dest.getDefiningOp<bufferization::AllocTensorOp>()) {
```

Comment:
You probably also want to check that the alloc tensor has shared memory space for this?